static/js/directives/ui/quota-management-view.js

```diff
@@ -76,139 +92,338 @@ angular.module('quay').directive('quotaManagementView', function () {
-	  $scope.newLimitConfig['limit_percent'] = null;
-	  loadOrgQuota();
-	});
+        ApiService.createOrganizationQuotaLimit(data, params).then((resp) => {
+          if (!rejectLimitExists()) {
+            $scope.newLimitConfig = {"limit_percent": 100, "type": $scope.rejectLimitType};
+          } else {
+            $scope.newLimitConfig['type'] = null;
+            $scope.newLimitConfig['limit_percent'] = null;
+          }
+          loadOrgQuota();
+        });
       }
 
-      $scope.updateQuotaLimit = function(limitId) {
-	var params = {
+      $scope.updateQuotaLimit = function (limitId) {
+        var params = {
           'orgname': $scope.organization.name,
           'quota_id': $scope.currentQuotaConfig.id,
           'limit_id': limitId,
-	};
+        };
 
-	var data = {
-	  'type': $scope.currentQuotaConfig['limits'][limitId]['type'],
-	  'threshold_percent': $scope.currentQuotaConfig['limits'][limitId]['limit_percent'],
-	};
+        var data = {
+          'type': $scope.currentQuotaConfig['limits'][limitId]['type'],
+          'threshold_percent': $scope.currentQuotaConfig['limits'][limitId]['limit_percent'],
+        };
 
-	ApiService.changeOrganizationQuotaLimit(data, params).then((resp) => {
-	  $scope.prevQuotaConfig['limits'][limitId]['type'] = $scope.currentQuotaConfig['limits'][limitId]['type'];
-	  $scope.prevQuotaConfig['limits'][limitId]['limit_percent'] = $scope.currentQuotaConfig['limits'][limitId]['limit_percent'];
-	});
+        ApiService.changeOrganizationQuotaLimit(data, params).then((resp) => {
+          $scope.prevQuotaConfig['limits'][limitId]['type'] = $scope.currentQuotaConfig['limits'][limitId]['type'];
+          $scope.prevQuotaConfig['limits'][limitId]['limit_percent'] = $scope.currentQuotaConfig['limits'][limitId]['limit_percent'];
+        }, function (resp) {
+          handleError(resp);
+        }
+        );
       }
-      
-      $scope.deleteQuotaLimit = function(limitId) {
-        params = {
+
+      $scope.deleteQuotaLimit = function (limitId) {
+        const params = {
           'orgname': $scope.organization.name,
           'quota_id': $scope.currentQuotaConfig.id,
           'limit_id': limitId,
         }
 
         ApiService.deleteOrganizationQuotaLimit(null, params).then((resp) => {
           delete $scope.currentQuotaConfig['limits'][limitId];
-	  delete $scope.prevQuotaConfig['limits'][limitId];
+          delete $scope.prevQuotaConfig['limits'][limitId];
+          populateDefaultQuotaLimits();
+        }, function (resp) {
+          handleError(resp);
         });
       }
 
-      $scope.disableSaveQuota = function() {
+      let displayErrorAlert = function (message) {
+        bootbox.alert(message);
+      }
+
+      let handleError = function (resp) {
+        if (resp.status == 403) {
+          displayErrorAlert("You do not have sufficient permissions to perform the action.");
+        } else {
+          displayErrorAlert(resp.data.error_message);
```

Comment:
should the possibility of a 500 error show up in a alert message or redirect here?

Reply:
All 500s get redirected to a different page (similar to - https://quay.io/500). So, they won't show up here.

Reply:
what about other error types perhaps 500 was the wrong  example

Reply:
They will be alerted based on the `error_message` returned.